doc/users/interactive_guide.rst

```diff
@@ -0,0 +1,432 @@
+into Matplotlib's :ref:`event handling system
+<event-handling-tutorial>`.  You can then use
+`.FigureCanvasBase.mpl_connect` to connect your function to
+Matplotlib's event handling system.  This allows you to directly
+interact with your data and write GUI toolkit agnostic user
+interfaces.
+
+
+.. _cp_integration:
+
+Command Prompt Integration
+==========================
+
+So far, so good.  We have the REPL (like the IPthon terminal) that
+lets us interactively send things code to the interpreter and get
+results back.  We also have the GUI toolkit that run an event loop
+waiting for user input and let up register functions to be run when
+that happens.  However, if we want to do both we have a problem: the
+prompt and the GUI event loop are both infinite loops that each think
+*they* are in charge!  In order for both the prompt and the GUI widows
+to be responsive we need a method to allow the loops to 'timeshare' :
+
+1. let the GUI main loop block the python process when you want
+   interactive windows
+2. let the CLI main loop block the python process and intermittently
+   run the GUI loop
+3. fully embed python in the GUI (but this is basically writing a full
+   application)
+
+.. _cp_block_the_prompt:
+
+Blocking the Prompt
+-------------------
+
+.. autosummary::
+   :template: autosummary.rst
+   :nosignatures:
+
+   pyplot.show
+   pyplot.pause
+
+   backend_bases.FigureCanvasBase.start_event_loop
+   backend_bases.FigureCanvasBase.stop_event_loop
+
+
+The simplest "integration" is to start the GUI event loop in
+'blocking' mode and take over the CLI.  While the GUI event loop is
+running you can not enter new commands into the prompt (your terminal
+may echo the charters typed into standard in, but they will not be
+sent to the Python interpreter because it is busy running the GUI
+event loop), but the figure windows will be responsive.  Once the
+event loop is stopped (leaving any still open figure windows
+non-responsive) you will be able to use the prompt again.  Re-starting
+the event loop will make any open figure responsive again (and will
+process and queued up user interaction).
+
+To start the event loop until all open figures are closed use
+`.pyplot.show` as ``pyplot.show(block=True)``.  To start the event
+loop for a fixed amount of time (in seconds) use `.pyplot.pause`.
+
+If you are not using `.pyplot` you can start and stop the event loops
+via `.FigureCanvasBase.start_event_loop` and
+`.FigureCanvasBase.stop_event_loop`. However, in most contexts where
+you would not be using `.pyplot` you are embedding Matplotlib in a
+large GUI application and the GUI event loop should already be running
+for the application.
+
+Away from the prompt, this technique can be very useful if you want to
+write a script that pauses for user interaction, see
+:ref:`interactive_scripts`.
+
+.. _spin_event_loop:
+
+Explicitly spinning the Event Loop
+----------------------------------
```

Comment:
May be nice to give a practical example or two at this point -- usually by the time I reach something like this I just switch to full embedding in a native (qt, in my case) app, so it's not totally clear to me what would be the use case here.

Reply:
Most of the content that was here got moved down to the scripts section, just moved the whole section down.